javalib/src/main/scala/java/util/AbstractSequentialList.scala

```diff
@@ -0,0 +1,42 @@
+package java.util
+
+abstract class AbstractSequentialList[E] protected ()
+    extends AbstractList[E] {
+
+  def get(index: Int): E =
+    listIterator(index).next
+
+  override def set(index: Int, element: E): E = {
+    val iter = listIterator(index)
+    val ret = iter.next
+    iter.set(element)
+    ret
+  }
+
+  override def add(index: Int, element: E): Unit =
+    listIterator(index).add(element)
+
+  override def remove(index: Int): E = {
+    val iter = listIterator(index)
+    val ret = iter.next
+    iter.remove
+    ret
+  }
+
+  override def addAll(index: Int, c: Collection[_ <: E]): Boolean = {
+    val iter = listIterator(index)
+    val citer = c.iterator
+    var changed = false
+    while (citer.hasNext) {
+      iter.add(citer.next)
+      iter.next
+      changed = true
+    }
+    changed
```

Comment:
with a ```for``` loop you need to convert from java to scala ```Iterator```
does not make sense in this case...
I try to stay with the original while loop